wrapper/src/main/java/software/amazon/jdbc/plugin/federatedauth/OktaAuthPlugin.java

```diff
@@ -0,0 +1,235 @@
+  public static AwsWrapperProperty
+      IDP_NAME = new AwsWrapperProperty("idpName", "okta", "The name of the Identity Provider implementation used");
+  public static final AwsWrapperProperty DB_USER =
+      new AwsWrapperProperty("dbUser", null, "The database user used to access the database");
+
+  private static final Logger LOGGER = Logger.getLogger(OktaAuthPlugin.class.getName());
+
+  protected final PluginService pluginService;
+
+  protected final RdsUtils rdsUtils;
+  protected final SamlUtils samlUtils;
+  private final IamTokenUtility iamTokenUtility;
+  private final TelemetryFactory telemetryFactory;
+  private final TelemetryGauge cacheSizeGauge;
+  private final TelemetryCounter fetchTokenCounter;
+
+  public OktaAuthPlugin(PluginService pluginService, CredentialsProviderFactory credentialsProviderFactory) {
+    this(pluginService, credentialsProviderFactory, new RdsUtils(), IamAuthUtils.getTokenUtility());
+  }
+
+  OktaAuthPlugin(
+      final PluginService pluginService,
+      final CredentialsProviderFactory credentialsProviderFactory,
+      final RdsUtils rdsUtils,
+      final IamTokenUtility tokenUtils) {
+    try {
+      Class.forName("software.amazon.awssdk.services.sts.model.AssumeRoleWithSamlRequest");
+    } catch (final ClassNotFoundException e) {
+      throw new RuntimeException(Messages.get("SamlAuthPlugin.javaStsSdkNotInClasspath"));
+    }
+
+    this.pluginService = pluginService;
+    this.credentialsProviderFactory = credentialsProviderFactory;
+    this.rdsUtils = rdsUtils;
+    this.samlUtils = new SamlUtils(this.rdsUtils);
+    this.iamTokenUtility = tokenUtils;
+    this.telemetryFactory = pluginService.getTelemetryFactory();
+    this.cacheSizeGauge = telemetryFactory.createGauge("oktaAuth.tokenCache.size", () -> (long) tokenCache.size());
+    this.fetchTokenCounter = telemetryFactory.createCounter("oktaAuth.fetchToken.count");
+  }
+
+  @Override
+  public Set<String> getSubscribedMethods() {
+    return Collections.unmodifiableSet(new HashSet<String>() {
+      {
+        add("connect");
+        add("forceConnect");
+      }
+    });
+  }
+
+  @Override
+  public Connection connect(String driverProtocol, HostSpec hostSpec, Properties props,
+      boolean isInitialConnection, JdbcCallable<Connection, SQLException> connectFunc) throws SQLException {
+    return connectInternal(hostSpec, props, connectFunc);
+  }
+
+  @Override
+  public Connection forceConnect(String driverProtocol, HostSpec hostSpec, Properties props,
+      boolean isInitialConnection, JdbcCallable<Connection, SQLException> forceConnectFunc)
+      throws SQLException {
+    return connectInternal(hostSpec, props, forceConnectFunc);
+  }
+
+  private Connection connectInternal(final HostSpec hostSpec, final Properties props,
+      final JdbcCallable<Connection, SQLException> connectFunc) throws SQLException {
+
+    this.samlUtils.checkIdpCredentialsWithFallback(IDP_USERNAME, IDP_PASSWORD, props);
```

Comment:
I think we need to open telemetry span here. It gives a user a good understanding of authentication timing.